router_request.c

```diff
@@ -317,15 +317,15 @@ router_request_reverse_all(router_t *router, router_request_t *req, router_reque
     } else {
         best_sp_index = (req->arrive_by ? req->from_stop_point : req->to_stop_point);
         do {
-            best_time = router->states_time[round * router->tdata->n_stop_points + best_sp_index];
+            best_time = router->states_time[(unsigned int) round * router->tdata->n_stop_points + best_sp_index];
```

Comment:
That doesn't solve the warning that `round` (a signed variable) is used in an array index calculation (which must always result in an unsigned value)